src/engine/strat_engine/liminal/liminal.rs

```diff
@@ -157,49 +175,74 @@ impl LiminalDevices {
         Ok(unlocked)
     }
 
-    /// Start a pool, create the devicemapper devices, and return the fully constructed
-    /// pool.
-    pub fn start_pool(
+    pub fn start_pool_legacy(
         &mut self,
-        pools: &Table<PoolUuid, StratPool>,
-        id: PoolIdentifier<PoolUuid>,
+        pools: &Table<PoolUuid, AnyPool>,
+        pool_uuid: PoolUuid,
         unlock_method: Option<UnlockMethod>,
-    ) -> StratisResult<(Name, PoolUuid, StratPool, Vec<DevUuid>)> {
-        let pool_uuid = match id {
-            PoolIdentifier::Uuid(u) => u,
-            PoolIdentifier::Name(n) => self
-                .name_to_uuid
-                .get(&n)
-                .ok_or_else(|| StratisError::Msg(format!("Could not find a pool with name {n}")))
-                .and_then(|uc| uc.to_result())?,
-        };
-        // Here we take a reference to entries in stopped pools because the call to unlock_pool
-        // below requires the pool being unlocked to still have its entry in stopped_pools.
-        // Removing it here would cause an error.
-        let encryption_info = self
+        passphrase_fd: Option<RawFd>,
+    ) -> StratisResult<(Name, PoolUuid, AnyPool, Vec<DevUuid>)> {
+        fn start_pool_failure(
+            pools: &Table<PoolUuid, AnyPool>,
+            pool_uuid: PoolUuid,
+            luks_info: StratisResult<Option<PoolEncryptionInfo>>,
+            infos: &HashMap<DevUuid, LStratisDevInfo>,
+            bdas: HashMap<DevUuid, BDA>,
+            meta_res: StratisResult<(DateTime<Utc>, PoolSave)>,
+        ) -> BDARecordResult<(Name, AnyPool)> {
+            let (timestamp, metadata) = match meta_res {
+                Ok(o) => o,
+                Err(e) => return Err((e, bdas)),
+            };
+
+            setup_pool_legacy(
+                pools, pool_uuid, luks_info, infos, bdas, timestamp, metadata,
+            )
+        }
+
+        let pool = self
             .stopped_pools
             .get(&pool_uuid)
             .or_else(|| self.partially_constructed_pools.get(&pool_uuid))
             .ok_or_else(|| {
                 StratisError::Msg(format!(
                     "Requested pool with UUID {pool_uuid} was not found in stopped or partially constructed pools"
                 ))
-            })?
-            .encryption_info();
-        let unlocked_devices = match (encryption_info, unlock_method) {
-            (Ok(Some(_)), None) => {
+            })?;
+
+        // Here we take a reference to entries in stopped pools because the call to unlock_pool
+        // below requires the pool being unlocked to still have its entry in stopped_pools.
+        // Removing it here would cause an error.
+        let encryption_info = pool.encryption_info();
+        let unlocked_devices = match (encryption_info, unlock_method, passphrase_fd) {
+            (Ok(Some(_)), None, _) => {
```

Comment:
The proposed change is much like the other one suggested. Start the match with:

```
(Ok(None), None, None) => Vec::new(),
(Ok(None), _, _) => {
    return Err(StratisError::Msg(format!(
                    "Pool with UUID {pool_uuid} is not encrypted but some unlock method was provided"
                )));    
}
```

And remove the matches now at line 223, and between 235 and 244. This aligns the code with my other suggestion and also removes a dependency on the ordering of the conditions of unlock method set and passphrase fd present by combining them into one.

Reply:
This makes the error message less specific. Can you explain a little bit more why this is desirable?

Reply:
This caught my attention because for every test, it's being done on legacy pools and new pools at the same time. Both situations, passphrase + method can happen at once, and, due to a different ordering between the code for the new and legacy pools, when both were set, the legacy pool error message picked the opposite thing to report on from the new pool error message. In itself, that is not so important, and might not have been noticed if we were not doing testing that way.

The thing I'm worrying about is future possibility of multiple token slots + the fact that the CLI user is likely to get this message. Given that there's potential for even more variety of ways to specify a way to start an encypted pool in future, all of which will not make sense for an unencrypted pool, I was thinking it would be helpful to just put them all in a single bucket.